pkg/inventory/inventory-client.go

```diff
@@ -441,33 +464,47 @@ func (cic *ClusterInventoryClient) updateStatus(obj *unstructured.Unstructured,
 	if !found {
 		return nil
 	}
-
-	klog.V(4).Infof("update inventory status")
 	mapping, err := cic.mapper.RESTMapping(obj.GroupVersionKind().GroupKind())
+	if err != nil {
+		return nil
+	}
+	hasStatus, err := cic.hasSubResource(obj.GetAPIVersion(), mapping.Resource.Resource, "status")
 	if err != nil {
 		return err
 	}
+	if !hasStatus {
+		klog.V(4).Infof("skip updating inventory status")
+		return nil
+	}
+
+	klog.V(4).Infof("update inventory status")
 	resource := cic.dc.Resource(mapping.Resource).Namespace(obj.GetNamespace())
 	meta := metav1.TypeMeta{
 		Kind:       obj.GetKind(),
 		APIVersion: obj.GetAPIVersion(),
 	}
-	liveObj, err := resource.Get(context.TODO(), obj.GetName(), metav1.GetOptions{TypeMeta: meta}, "status")
-	if err != nil {
-		if apierrors.IsNotFound(err) {
-			klog.V(4).Infof("skip updating inventory subresource status for %v: %v", meta, err)
-			return nil
-		}
-		return fmt.Errorf("failed to get inventory status from cluster: %w", err)
-	}
-
-	err = unstructured.SetNestedMap(liveObj.UnstructuredContent(), status, "status")
-	if err != nil {
+	// set the resourceVersion of the inventory object
+	obj.SetResourceVersion(cic.invResourceVersion)
```

Comment:
Is UID not also checked? Usually it's required to perform an update, at least on normal resources.

Kernel team says if subresource update doesn't check the UID it might be a bug.

Reply:
How could we check the UID if there is no extra `Get` to get the live object from the cluster? The UID in `liveObj` is from the last Get/Update/Create call.

Reply:
I think we should be getting the UID from Storage.GetObject().

However, it looks like we don't actually need it, because it's not actually required to be the same on update, so it won't error as-is. If we wanted it to error we'd have to specify an UpdateOption.Precondition. But I don't know if it really matters. If something else is recreating inventory asynchronously, it's probably gonna break in other ways too.